test/functional/wallet.py

```diff
@@ -144,6 +150,10 @@ def run_test(self):
         assert_equal(self.nodes[2].getbalance(), 94)
         assert_equal(self.nodes[2].getbalance("from1"), 94-21)
 
+        # Verify that a spent output cannot be locked anymore
+        spent_0 = {"txid": node0utxos[0]["txid"], "vout": node0utxos[0]["vout"]}
+        assert_raises_jsonrpc(-8, "Invalid parameter, expected unspent output", self.nodes[2].lockunspent, False, [spent_0])
```

Comment:
This is the bug. `lockunspent()` should be called on `self.nodes[0]`, not `self.nodes[2]`

Reply:
Thanks! This one stumped me because for some reason when I ran the wallet tests locally, they didn't fail..

Reply:
Yes! I saw the same thing. Very odd. h/t @sdaftuar for spotting the bug.